src/plugins/I18nPlugin.ts

```diff
@@ -137,14 +112,65 @@ export default class DojoI18nPlugin {
 			});
 		}
 
-		compiler.plugin('compilation', (compilation: any) => {
+		compiler.plugin('compilation', (compilation: any, data: any) => {
+			const astMap = new Map();
+			const containsLoad: string[] = [];
+
+			data.normalModuleFactory.plugin('before-resolve', (result: any, callback: any) => {
+				if (!result) {
+					return callback();
+				}
+
+				let request = isRelative(result.request) ? path.join(result.context, result.request) : result.request;
+
+				const { contextInfo } = result;
+				const issuer = contextInfo && contextInfo.issuer;
+				if (issuer && issuer.indexOf('node_modules') < 0 && isCldrLoadModule(request)) {
+					containsLoad.push(issuer);
+				}
+
+				return callback(null, result);
+			});
+
+			data.normalModuleFactory.plugin('parser', (parser: any) => {
+				parser.plugin('program', (ast: any) => {
+					const path = parser.state.current.userRequest;
+					if (path) {
+						astMap.set(path, ast);
```

Comment:
Since it cannot be determined whether a module requires `@dojo/i18n/cldr/load` until after it has been parsed, the AST objects are stored in the map indiscriminately. For large applications, I wonder whether the memory consumption would be problematic. However, I have yet to find a reliable means of tracking when a module's dependencies have been completely parsed to remove that module's AST object from the cache. If anyone has any suggestions about managing this, I'd greatly appreciate it.